examples/robot_localization_failure/run.py

```
@@ -1,103 +1,311 @@
+            width_meters=config.architecture.width_meters,
+        ),
+        dataset_kwargs={
+            "image_size": config.architecture.image_size,
+            "width_meters": config.architecture.width_meters,
+        },
     )
-    | Drop("/scan", "/particle_cloud", "/momo/pose", "/amcl_pose")
-    # detect experiment slice points based on delocalization events
-    | DetectDelocalizations("/delocalizations", name="slice_points")
-    | Drop("/delocalizations")
-    | SliceTimeSeries(
-        time_series="measurements",
-        slice_points="slice_points",
+    model = learner.learn(
+        inputs=train_data.drop(["is_delocalized"]),
+        outputs=train_data.select(["is_delocalized"]),
     )
-    | Drop("slice_points")
-    # detect localization status based on position and heading errors
-    | LocalizationStatus(
-        time_series="measurements",
-        ground_truth="/momo/pose",
-        estimation="/amcl_pose",
-        position_threshold=config.localization.position_threshold,
-        heading_threshold=config.localization.heading_threshold,
+
+    # Evaluate the model
+    metrics = [
+        Accuracy(),
+        ClassificationReport(),
+        FBetaScore(beta=1.0),
+        PrecisionScore(),
+        Recall(),
+    ]
+    report = evaluate_offline(
+        model,
+        DataFrame(test_data),
+        inputs=test_data.drop(["is_delocalized"]).columns,
+        outputs=["is_delocalized"],
+        metrics=metrics,
     )
-)
+    print(report)
+    # Save the model
+    out_path = f"models/{rosbag_dir.name}.pt"
+    msg = f"Saving model to {out_path}"
+    logger.info(msg)
+    torch.save(model.module.state_dict(), out_path)
+
+
+def get_rosbag_paths(
+    config: DictConfig | ListConfig,
+) -> tuple[Path, list[Path]]:
+    rosbag_dir = Path(config.rosbag.path)
+
+    if not rosbag_dir.is_dir():
+        msg = f"Specified ROS2 bag directory {rosbag_dir} does not exist."
+        logger.error(msg)
+        sys.exit(1)
```

raise an exception instead of log and exit